pandaserver/dataservice/closer.py

```diff
@@ -0,0 +1,465 @@
+"""
+update dataset DB, and then close dataset and start Activator if needed
+
+"""
+
+import re
+import sys
+import datetime
+
+from typing import Dict, List
+
+from pandacommon.pandalogger.PandaLogger import PandaLogger
+from pandacommon.pandalogger.LogWrapper import LogWrapper
+
+from pandaserver.config import panda_config
+from pandaserver.dataservice import Notifier
+from pandaserver.dataservice.Activator import Activator
+from pandaserver.taskbuffer import EventServiceUtils
+
+# logger
+_logger = PandaLogger().getLogger("closer")
+
+
+def init_logger(p_logger: PandaLogger) -> None:
+    """
+    Redirect logging to parent as it doesn't work in nested threads
+
+    Args:
+        p_logger (PandaLogger): The parent logger.
+    """
+    # redirect logging to parent as it doesn't work in nested threads
+    global _logger
+    _logger = p_logger
+    Notifier.initLogger(_logger)
+
+
+class Closer:
+    """
+    Update dataset DB, and then close dataset and start Activator if needed
```

Comment:
I don't understand this comment (also CCing @tmaeno) - what is the relation between Closer and Activator? I'm also looking at https://panda-wms.readthedocs.io/en/latest/architecture/server.html#other-panda-modules to try to get a general picture. (After all the refactoring we should eventually also update the readthedocs)

DB -> dispatch block, or are with updating the dataset in the DB?